src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -45,11 +47,14 @@ private void AddExceptionDiagnostic(Diagnostic diagnostic)
         /// </summary>
         public async Task<ImmutableArray<Diagnostic>> GetAnalyzerDiagnosticsAsync()
         {
+            _driver.StartCompleteAnalysis(_analyzedSyntaxTrees, _cancellationToken);
+
             // Invoke GetDiagnostics to populate the compilation's CompilationEvent queue.
             // Discard the returned diagnostics.
             _compilation.GetDiagnostics(_cancellationToken);
```

Comment:
I don't think any compilation events will have been processed at this point, except those generated (and already consumed) for analyzing individual documents.